bin/fuel-core/src/cli/snapshot.rs

```diff
@@ -53,46 +78,110 @@ pub async fn exec(command: Command) -> anyhow::Result<()> {
-    use anyhow::Context;
-    use fuel_core::{
-        chain_config::{
-            ChainConfig,
-            StateConfig,
-        },
-        database::Database,
-    };
-    let path = command.database_path;
-    let data_source = fuel_core::state::rocks_db::RocksDb::default_open(&path, None)
-        .map_err(Into::<anyhow::Error>::into)
-        .context(format!(
-            "failed to open database at path {}",
-            path.display()
-        ))?;
-    let db = Database::new(std::sync::Arc::new(data_source));
+pub fn exec(command: Command) -> anyhow::Result<()> {
+    let db = open_db(&command.database_path)?;
 
     match command.subcommand {
         SubCommands::Everything {
             chain_config,
             output_dir,
-        } => {
-            let chain_conf = match chain_config.as_deref() {
-                None => ChainConfig::local_testnet(),
-                Some(path) => ChainConfig::load_from_directory(path)?,
-            };
-            let state_conf = StateConfig::generate_state_config(db)?;
-
-            std::fs::create_dir_all(&output_dir)?;
-            chain_conf.create_config_file(&output_dir)?;
-            state_conf.create_config_file(&output_dir)?;
-        }
-        SubCommands::Contract { contract_id } => {
-            let config = db.get_contract_config_by_id(contract_id)?;
-            let stdout = std::io::stdout().lock();
-
-            serde_json::to_writer_pretty(stdout, &config)
-                .context("failed to dump contract snapshot to JSON")?;
-        }
+            state_encoding_format,
+        } => full_snapshot(chain_config, &output_dir, state_encoding_format, &db),
+        SubCommands::Contract { contract_id } => contract_snapshot(&db, contract_id),
     }
+}
+
+fn contract_snapshot(
+    db: impl ChainStateDb,
+    contract_id: ContractId,
+) -> Result<(), anyhow::Error> {
+    let config = db.get_contract_config_by_id(contract_id)?;
+    let stdout = std::io::stdout().lock();
+    serde_json::to_writer_pretty(stdout, &config)
+        .context("failed to dump contract snapshot to JSON")?;
     Ok(())
 }
+
+fn full_snapshot(
+    chain_config: Option<PathBuf>,
+    output_dir: &Path,
+    state_encoding_format: StateEncodingFormat,
+    db: impl ChainStateDb,
+) -> Result<(), anyhow::Error> {
+    let encoder = initialize_encoder(output_dir, state_encoding_format)?;
+    write_chain_state(db, encoder)?;
+
+    let chain_config = load_chain_config(chain_config)?;
+    chain_config.create_config_file(output_dir)?;
+
+    Ok(())
+}
+
+fn write_chain_state(db: impl ChainStateDb, mut encoder: Encoder) -> anyhow::Result<()> {
+    fn write<T>(
+        data: impl Iterator<Item = StorageResult<T>>,
+        group_size: usize,
+        mut write: impl FnMut(Vec<T>) -> anyhow::Result<()>,
+    ) -> anyhow::Result<()> {
+        data.chunks(group_size)
+            .into_iter()
+            .try_for_each(|chunk| write(chunk.try_collect()?))
+    }
+
+    let group_size = 1000;
```

Comment:
For `parquet` the group size gets baked into each group so that during decoding we don't need to know the group size.

For `json` we decided against encoding group sizes so we'd have a simpler schema for users (also fuel-core tests and the SDKs). When decoding `json` you need to specify the group sizes and they will be created on the fly.

I'll add the group size as a CLI parameter. Ideally, make it available only if the encoding is `parquet`.

Reply:
Added the group size as well as the encoding to the CLI.